compiling error of the master branch

with Dune 2.3 and no mpi. 


```
 In file included from /home/kaib/OPM-master-test/build/opm-autodiff/examples/flow.cpp:82:0:
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp: In function ‘void Opm::distributeGridAndData(Dune::CpGrid&, Opm::DeckConstPtr, Opm::EclipseStateConstPtr, Opm::BlackoilStat
e&, Opm::BlackoilPropsAdFromDeck&, Opm::DerivedGeology&, std::shared_ptr<Opm::EclMaterialLawManager<Opm::ThreePhaseMaterialTraits<double, 0, 1, 2> > >&, boost::any&, bool)’:
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:358:13: error: ‘ParallelIndexSet’ in ‘class Dune::CpGrid’ does not name a type
     typedef Dune::CpGrid::ParallelIndexSet IndexSet;
             ^
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:359:11: error: ‘IndexSet’ does not name a type
     const IndexSet& local_indices  = grid.getCellIndexSet();
           ^
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:360:24: error: ‘local_indices’ was not declared in this scope
     for ( auto index : local_indices )
                        ^
```

`flow`.  I have dune-cornerpoint, let me check if I have dune-grid. I think I should have dune-grid. My configuration has been working for long time. The last time I tried to compile the master was this Monday. 

```
In file included from /home/kaib/OPM-master-test/build/opm-autodiff/examples/flow.cpp:82:0:
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp: In function ‘void Opm::distributeGridAndData(Dune::CpGrid&, Opm::DeckConstPtr, Opm::EclipseStateConstPtr, Opm::BlackoilStat
e&, Opm::BlackoilPropsAdFromDeck&, Opm::DerivedGeology&, std::shared_ptr<Opm::EclMaterialLawManager<Opm::ThreePhaseMaterialTraits<double, 0, 1, 2> > >&, boost::any&, bool)’:
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:358:13: error: ‘ParallelIndexSet’ in ‘class Dune::CpGrid’ does not name a type
     typedef Dune::CpGrid::ParallelIndexSet IndexSet;
             ^
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:359:11: error: ‘IndexSet’ does not name a type
     const IndexSet& local_indices  = grid.getCellIndexSet();
           ^
/home/kaib/OPM-master-test/build/opm-autodiff/opm/autodiff/RedistributeDataHandles.hpp:360:24: error: ‘local_indices’ was not declared in this scope
     for ( auto index : local_indices )
                        ^
```

--- a/opm/autodiff/RedistributeDataHandles.hpp
+++ b/opm/autodiff/RedistributeDataHandles.hpp
@@ -45,7 +45,7 @@ inline void distributeGridAndData( Grid& ,
 {
 }
 
-#if HAVE_DUNE_CORNERPOINT
+#if HAVE_DUNE_CORNERPOINT && HAVE_MPI
 /// \brief a data handle to distribute Derived Geology
 class GeologyDataHandle
 {

